lib/algos/list/Extension.js

```diff
@@ -23,6 +23,32 @@ class Extension {
         this.keys = 0;
     }
 
+    /**
+     * Filters-out non-requested optional fields from the value. This function
+     * shall be applied on any value that is to be returned as part of the
+     * result of a listing extension.
+     *
+     * @param {Object} value - The JSON value of a listing item
+     *
+     * @return {Object} The value without the filtered-out fields.
+     */
+    filterContent(value) {
+        let ret = undefined;
+        try {
+            ret = JSON.parse(value);
+            delete ret.location;
+            ret = JSON.stringify(ret);
+        } catch (e) {
+            this.logger.warn('Could not parse Object Metadata while listing',
```

Comment:
If deemed necessary, we could protect against an undefined logger here with a conditional construct.
Wasn't sure it'd be worth it, so just mentionning it here.